### PR TITLE
[QPG] Fix shell app

### PR DESCRIPTION
**Pull request by lucicop**
Problem

Shell app build fails for QPG.

Change overview

fix shell app failure by defining necessary parameter
move CHIP Project config files to each example folder (needed to define a PID based on the application type)
add Shell app to CI
Testing

The changes were tested by compiling each app (lighting-app, lock-app, persistent-storage, shell)

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

File: .github/workflows/examples-qpg.yaml [hunk 1]
```diff
@@ -76,6 +76,14 @@ jobs:
                     qpg qpg6105+debug lighting-app \
                     out/lighting_app_debug/chip-qpg6105-lighting-example.out \
                     /tmp/bloat_reports/
+            - name: Build example QPG6105 Shell App
+              timeout-minutes: 5
+              run: |
+                  scripts/examples/gn_build_example.sh examples/shell/qpg out/shell_app_debug qpg_target_ic=\"qpg6105\"
```
[the remark this answers is not in the document]
Comment (lucicop) [reply .github/workflows/examples-qpg.yaml:82]: I will create a Github issue for this, would prefer to finalize the merge before something else breaks it. The CI update can be done separately, in my view.

[the remark this answers is not in the document]
Comment (lucicop) [reply .github/workflows/examples-qpg.yaml:82]: I am preparing the changes and will add them as part of this PR.

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**